java/src/jmri/jmris/srcp/JmriSRCPThrottleServer.java

```diff
@@ -331,14 +331,8 @@ public void setThrottleFunctions(int bus, int address, ArrayList<Boolean> fList)
                 log.debug("Throttle in throttle list");
                 Throttle t = throttleList.get(addressList.indexOf(addr));
                 for (int i = 0; i < fList.size(); i++) {
-                    try {
-                        java.lang.reflect.Method setter = t.getClass()
-                                .getMethod("setF" + i, boolean.class);
-                        setter.invoke(t, fList.get(i));
-                    } catch (java.lang.NoSuchMethodException
-                            | java.lang.IllegalAccessException
-                            | java.lang.reflect.InvocationTargetException ex1) {
-                        log.error("", ex1);
+                    t.setFunction(i, fList.get(i));
+                    if ( i > t.getFunctions().length-1) {
```

Comment:
This sets the function value and then ALWAYS sends an error status.  That isn't right.

The If statement needs to be before the `t.setFunction()` call with the send of error status as the else.

Reply:
See prev. comment